openstack/objectstorage/v1/swauth/requests.go

```diff
@@ -37,7 +41,7 @@ func Auth(c *gophercloud.ProviderClient, opts AuthOptsBuilder) (r GetAuthResult)
 		}
 	}
 
-	resp, err := c.Request("GET", getURL(c), &gophercloud.RequestOpts{
+	resp, err := c.Request(context.Background(), "GET", getURL(c), &gophercloud.RequestOpts{
```

Comment:
Please avoid generating "root" context objects outside convenience functions.

Here you could create `AuthWithContext` that accepts a context, then add a convenience function `Auth` for backwards compatibility that calls `AuthWithContext` with a `context.Background()`.